cmd/utils/hierarchical_coordinator.go

```diff
@@ -1321,3 +1326,29 @@ func (hc *HierarchicalCoordinator) GetBackendForLocationAndOrder(location common
 	}
 	return nil
 }
+
+func ReIndexChainIndexer() {
+	providedDataDir := viper.GetString(DataDirFlag.Name)
+	if providedDataDir == "" {
+		log.Global.Fatal("Data directory not provided for reindexing")
+	}
+	dbDir := filepath.Join(filepath.Join(providedDataDir, "zone-0-0/go-quai"), "chaindata")
```

Comment:
Please include nodeLocation as an argument here

Reply:
There's no nodelocation here, it's in the hierarchical coordinator

Reply:
As this isn't part of consensus and is optional, I'm fine with not building the multi-chain aspect.